pkg/cmd/browse/browse.go

```diff
@@ -0,0 +1,245 @@
+package browse
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type browser interface {
+	Browse(string) error
+}
+
+type BrowseOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Browser    browser
+
+	SelectorArg   string
+	AdditionalArg string
+
+	ProjectsFlag bool
+	WikiFlag     bool
+	SettingsFlag bool
+	BranchFlag   bool
+}
+
+type exitCode int
+
+const (
+	exitUrlSuccess    exitCode = 0
+	exitNonUrlSuccess exitCode = 1
+	exitNotInRepo     exitCode = 2
+	exitTooManyFlags  exitCode = 3
+	exitTooManyArgs   exitCode = 4
+	exitExpectedArg   exitCode = 5
+	exitInvalidCombo  exitCode = 6
+)
+
+func NewCmdBrowse(f *cmdutil.Factory) *cobra.Command {
+
+	opts := &BrowseOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Browser:    f.Browser,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Long:  "Open specific pages in the browser on GitHub",
```

Comment:
How about:
```suggestion
		Long:  "Open the GitHub repository in the web browser.",
```
Rationale:
- Just "browser" can be generic, but "web browser" narrows it down
- We don't need to mention "specific pages" since the default mode (with no arguments) just opens the repository homepage.